### ISSUE TITLE
/bundle/gems/activerecord-5.2.0/lib/active_record/log_subscriber.rb:98 is everywhere.

### ISSUE THREAD

**Issue by ghost** [rails/rails#32935]
### Steps to reproduce

I simply start a default new app.

### Expected behavior
Not should happen what instead happens.

### Actual behavior
I sse this line everywhere:

`↳ /bundle/gems/activerecord-5.2.0/lib/active_record/log_subscriber.rb:98`

For example:

```
Started GET "/" for 127.0.0.1 at 2018-05-19 16:14:17 +0000
Processing by RootController#index as HTML
  User Load (0.7ms)  SELECT  "users".* FROM "users" WHERE "users"."id" = $1 ORDER BY "users"."id" ASC LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ /bundle/gems/activerecord-5.2.0/lib/active_record/log_subscriber.rb:98
Completed 302 Found in 7ms (ActiveRecord: 0.7ms)
```

### System configuration
**Rails version**: 5.2.0
**Ruby version**: 2.5.1

**Comment by utilum** [comment]
I can't reproduce this locally. Are you sure it's an app as-generated?

```
]$ rails -v
Rails 5.2.0
]$ ruby -v
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
]$ rails s
=> Booting Puma
=> Rails 5.2.0 application starting in development 
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.11.4 (ruby 2.5.1-p57), codename: Love Song
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop
Started GET "/" for 127.0.0.1 at 2018-05-19 20:58:25 +0200
Processing by Rails::WelcomeController#index as HTML
  Rendering /home/u/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/templates/rails/welcome/index.html.erb
  Rendered /home/u/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/railties-5.2.0/lib/rails/templates/rails/welcome/index.html.erb (3.0ms)
Completed 200 OK in 11ms (Views: 5.6ms | ActiveRecord: 0.0ms)


^C- Gracefully stopping, waiting for requests to finish
=== puma shutdown: 2018-05-19 20:58:38 +0200 ===
- Goodbye!
Exiting
```

**Comment by rentalname** [comment]
I saw same message in my environment.

This message generate from this
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/log_subscriber.rb#L98
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/log_subscriber.rb#L111

**Comment by johnunclesam** [comment]
Yes, it is a new app with devise only. Maybe the same like this: https://github.com/rails/rails/issues/32719 ?

**Comment by utilum** [comment]
Could you provide a minimal app (just one simple use of active storage, without devise or other unnecessary extras) that reproduces this?

**Comment by BobGu** [comment]
I was also unable to reproduce the bug using the same ruby and rails versions using a vanilla new rails app.

**Comment by utilum** [comment]
Perhaps the problem rises between Rails and Devise?

@BobGu does your vanilla app include a model with an `ActiveStorage` attribute?

**Comment by RandieM** [comment]
Hello everyone,

I just noticed that I also have this line in my terminal. It seems to appear just after the SELECT on the schema_migrations. As I am not using a database (I did use one in the past, but switched to table-less models), it only appears once, as soon as I start the app. I am not using devise.

```Booting Puma
Rails 5.2.0 application starting in development 
Puma starting in single mode...
* Version 3.11.4 (ruby 2.4.4-p296), codename: Love Song
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://localhost:3003
Started GET "/" for ::1 at 2018-07-10 13:37:03 +0200
   (3.5ms)  SELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC
↳ /home/removed/.gem/ruby/gems/activerecord-5.2.0/lib/active_record/log_subscriber.rb:98

**Comment by kaspth** [comment]
Looks related to https://github.com/rails/rails/commit/3876defd7c951335129fa92c573c7b8fce085aac

@ghost @RandieM do either of your apps have `verbose_query_logs` enabled in new_framework_defaults_5_2.rb?

Strange that the offending call site it finds is the log_subscriber though. I wouldn't have expected it to log the query source for the schema migrations select either. cc @olivierlacan

**Comment by olivierlacan** [comment]
@kaspth It logs the query source for any SQL query, so schema_migrations showing up makes sense since it shows up in dev logs. 

@RandieM Can you give me more context about how you're running your apps locally? Docker? rbenv/RVM/chruby? A way to reproduce this issue in a new Rails app using the same Gemfile would be helpful.

I've seen some gems mess up the backtrace like this before but didn't get a lot of reports. It could be that few people are using verbose_query_logs or that few are affected. Not sure.

> As I am not using a database (I did use one in the past, but switched to table-less models), it only appears once, as soon as I start the app. I am not using devise.
@RandieM ActiveRecord is querying the `schema_migrations` table for you so Rails is using a database on your behalf. Is it possible that you may not have fully deactivated ActiveRecord?

**Comment by RandieM** [comment]
@kaspth, I do not have this enabled in new_framework_defaults_5_2.rb.

@olivierlacan, this is highly probable, which is also the reason why I mentioned the text that you quoted. As said, I was using a database in the past and had to switch to table-less models due to GDPR. Since I did not seem to find any "official" instructions on this, I just followed some guides found on tech blogs, where there was no information on deactivating ActiveRecord. Having a quick look, I now found [this thread](https://stackoverflow.com/questions/19078044/disable-activerecord-for-rails-4#19078854). Does the accepted answer, in your opinion, correctly describe the steps to fully deactivate ActiveRecord?
On the other hand, does the fact that I have not removed my DB schema etc justify this error/warning in my terminal or could this still be caused by something else?
I run the app locally either in rbenv or "directly" in my Linux home without any environment manager; the "error" appears in both cases.

**Comment by kaspth** [comment]
@olivierlacan I was wondering if it made sense to show where queries that the user didn't write came from. But yes, it does. It's better that the config is just all or nothing.

@RandieM if you have `require "rails/all"` in `config/boot.rb` that'll load Active Record, which might try to read schema_migrations since the database still exists. Anyway, that's probably the limit where we can help you without a reproduction in a generated demo app.

**Comment by RandieM** [comment]
@kaspth, thanks for your tip.

I found a few spare minutes and I managed to reproduce the behaviour in a fresh Rails app, in which I created a table and immediately dropped it via Rails migrations. Starting the app in development gave out the same message (`↳ /home/removed/.gem/ruby/gems/activerecord-5.2.0/lib/active_record/log_subscriber.rb:98`). Deleting the migration files made the message disappear.
For me, it is clear that there were further steps that I had to take after dropping my website's tables. 

What was not so clear, though, is why I even received this message, since according to log_subscriber.rb:98, the method is only called if verbose_query_logs is enabled, which I did not have in new_framework_defaults_5_2.rb. It turns out, however, that I had the line `config.active_record.verbose_query_logs = true` in development.rb, which explains everything.

**Comment by olivierlacan** [comment]
@RandieM Can you put that fresh Rails app on GitHub so I can get it locally to work on this? Thanks.

`verbose_query_logs` are enabled by default in new Rails 5.2 apps in development, so that's why you saw them.

It's not extremely surprising that occasionally it would report `LogSubscriber` as the originator of the query because this is where the SQL query is actually sent to the logger from. I need to ensure that we never report `LogSubscriber` though because it quite literally can never be the actual origin of the SQL call.

**Comment by RandieM** [comment]
@olivierlacan, sure. You can find it here: https://github.com/RandieM/log_subscriber_testapp.

In fact, I had to re-create the app (as I had already deleted files in my attempt to "get rid of" the message) and it looks that this time I didn't even have to drop the table, in order to receive the message, as I initially had thought.

**Comment by kaspth** [comment]
@olivierlacan someone already fixed that: https://github.com/rails/rails/pull/33455 😊

**Comment by y-yagi** [comment]
Because issue creator's account has been deleted and part of the issue has been solved with #33455, I close this.
